core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -89,6 +89,10 @@ pub(crate) trait StateContainer<Tx: TransactionWithMeta> {
     /// Returns `true` if a packet was dropped due to capacity limits.
     fn push_id_into_queue(&mut self, priority_id: TransactionPriorityId) -> bool;
 
+    /// A transaction ID was retrieved for scheduling, when it's not scheduled, put it back into
+    /// priority queue. "put back" shall never need to push out other tx (os is it?)
+    fn put_id_back_into_queue(&mut self, priority_id: TransactionPriorityId);
+
```

Comment:
On the second thought, not too sure if this is the right solution. Is Schedule *putting back* a ID really so different from receiver insert one? Or maybe need a 3rd state that if a transaction is retrieved for scheduling, it's space in Container shall be "reserved", making it not available for new received transactions, until its either scheduled or put-back.